quickmpc/utils/parse_csv.py

```diff
@@ -19,17 +20,17 @@ def format_check(secrets: List[List[float]],
 
 
 def to_float(val: str) -> float:
-    """ If val is a float, convert as is; if it is a string, unicode it. """
+    """ If val is a float, convert as is; if it is a string, hash it. """
     try:
         return float(val)
     except ValueError:
         # k,m are constants used in the comparison operation
         # Due to the limitation of comparison operation,
         # k bits are taken out and divided by 2^m.
-        k: int = 32
-        m: int = 16
-        val_bin: str = "".join(map(lambda x: bin(ord(x))[2:], val))
-        val_int: int = int(val_bin[:k], 2)
+        k: int = 48
+        m: int = 20
+        hs: str = sha512(val.encode()).hexdigest()
+        val_int: int = int(hs[:(k >> 2)], 16)
         val_float: float = val_int / pow(2, m)
```

Comment:
@mdonaka 
Is there any clear intention behind the shaky power of 2 notation in `k >> 2` and `pow(2, m)`?

Reply:
@come2ry 
I wasn't aware of this, but I would like to keep it that way.

`(k>>2)` is a divide by 4 operation, but it cannot be expressed in pow. (e.g., 12,20,...).
and I want the shift to be handled as an integer.

`pow(2,m)` can be expressed as a shift, but we want to keep it as pow to match Typescript.
Typescript uses pow because typescript breaks if the shift exceeds 32 bits.